File: pattern/pattern8/code.py
class Solution:
    def pattern8(self, n):
        for i in range(0,n):
            for j in range(0, i):
                print(" ", end = "")
            for j in range(0, 2 * (n - i) -1):
                print("*", end = "")
            print()
            

File: pattern/pattern7/code.java
class Solution {
    public void pattern7(int n) {
        for (int i = 1; i<=n; i++){
            for (int j = 0 ; j < n - i; j++){
                System.out.print(" ");
            }
            for (int j = 0; j < 2 * i - 1; j++){
                System.out.print("*");
            }
            for (int j = 0 ; j < n - i; j++){
                System.out.print(" ");
            }
            System.out.println();
        }
    }
}
